src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/FlowProcessingWithHistorySupportFsm.java

```diff
@@ -57,9 +59,7 @@ public void saveActionToHistory(String action) {
         sendHistoryData(action, null);
     }
 
-    /**
-     * Add a history record on the action.
-     */
+    @Override
```

Comment:
Should method above `public void saveActionToHistory(String action) {` be in interface too?

Reply:
No, at least at this moment. I would like to "move" this history interface anywhere from base FSM code, I have a lot of issues now, because I do not use same inheritance chain as other CRUD operations.

This API should be somewhere on carrier or service level.